### PR TITLE
Update code style to modern recommendations. Implement Closable.

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

File: tape/src/main/java/com/squareup/tape/QueueFile.java [hunk 1]
```diff
@@ -134,19 +130,7 @@ private static void writeInt(byte[] buffer, int offset, int value) {
     buffer[offset + 3] = (byte) value;
   }
 
-  /**
-   * Stores int values in buffer. The behavior is equivalent to calling {@link
-   * java.io.RandomAccessFile#writeInt} for each value.
-   */
-  private static void writeInts(byte[] buffer, int... values) {
```
Comment (JakeWharton) [inline tape/src/main/java/com/squareup/tape/QueueFile.java]: This method was wasteful. Caused `int[]` allocations and was always called with the same number of params.

File: tape/src/main/java/com/squareup/tape/QueueFile.java [hunk 2]
```diff
@@ -201,14 +186,16 @@ private static void initialize(File file) throws IOException {
       raf.setLength(INITIAL_LENGTH);
       raf.seek(0);
       byte[] headerBuffer = new byte[16];
-      writeInts(headerBuffer, INITIAL_LENGTH, 0, 0, 0);
+      writeInt(headerBuffer, 0, INITIAL_LENGTH);
```
Comment (JakeWharton) [inline tape/src/main/java/com/squareup/tape/QueueFile.java]: New byte arrays are already empty. No need to write empty slots.

File: tape/src/main/java/com/squareup/tape/QueueFile.java [hunk 3]
```diff
@@ -290,14 +277,14 @@ public void add(byte[] data) throws IOException {
...
-   * @param count  number of bytes to copy
-   * @throws IndexOutOfBoundsException if {@code offset < 0} or {@code count < 0}, or if {@code offset + count} is
-   *                                   bigger than the length of {@code buffer}.
+   * @param count number of bytes to copy
+   * @throws IndexOutOfBoundsException if {@code offset < 0} or {@code count < 0}, or if {@code
+   * offset + count} is bigger than the length of {@code buffer}.
    */
   public synchronized void add(byte[] data, int offset, int count) throws IOException {
-    nonNull(data, "buffer");
+    if (data == null) throw new NullPointerException("data == null");
```
Comment (f2prateek) [inline tape/src/main/java/com/squareup/tape/QueueFile.java]: Not sure where the conventions are from, but this should be made consistent with the other places as:

```
if (data == null) {
  throw new NullPointerException("data == null");
}
```

**Comment by f2prateek** [comment]
LGTM!

**Pull request merged**